app.js

```diff
@@ -2,10 +2,103 @@
 // this will check if we have a user and set signout link if it exists
 import './auth/user.js';
 
+import { createList, getList, boughtList, deleteAllBoughtLists } from './fetch-utils.js';
+
+import { renderList } from './render-utils.js';
+
 /* Get DOM Elements */
+const addItemForm = document.getElementById('add-item-form');
+const errorDisplay = document.getElementById('error-display');
+const itemList = document.getElementById('item-list');
+const deleteButton = document.getElementById('delete-button');
 
 /* State */
-
+let lists = [];
+let error = null;
 /* Events */
 
+addItemForm.addEventListener('submit', async (e) => {
+    e.preventDefault();
+    const formData = new FormData(addItemForm);
+    const newList = {
+        item: formData.get('item'),
+        quantity: formData.get('quantity'),
+    };
+
+    const response = await createList(newList);
+    error = response.error;
+    const list = response.data;
+
+    if (error) {
+        displayError();
+    } else {
+        lists.push(list);
+        displayLists();
+        addItemForm.reset();
+    }
+});
+
+window.addEventListener('load', async () => {
+    const response = await getList();
+
+    error = response.error;
+    lists = response.data;
+
+    if (error) {
+        displayError();
+    }
+    if (lists) {
+        displayLists();
+    }
+});
```

Comment:
just stylistically, it makes more sense for something that happens on page load to be written before writing what happens on form